src/main/java/com/igot/cb/discussion/service/impl/DiscussionServiceImpl.java

```diff
@@ -1216,16 +1220,32 @@ private String validateGetBookmarkedDiscussions(Map<String, Object> requestData)
         return errorMsg.toString();
     }
 
-    private List<Map<String, Object>> fetchAndEnhanceDiscussions(List<Map<String, Object>> discussions) {
+    private void fetchAndEnhanceDiscussions(List<Map<String, Object>> discussions, boolean isAnswerPost) {
         Map<String, String> discussionToCreatedByMap = discussions.stream()
                 .collect(Collectors.toMap(
                         discussion -> discussion.get(Constants.DISCUSSION_ID).toString(),
                         discussion -> discussion.get(Constants.CREATED_BY).toString()));
 
+        Map<String, List<String>> discussionToUserTagMap = new HashMap<>();
+        if (isAnswerPost) {
+            discussionToUserTagMap.putAll(discussions.stream()
+                    .filter(discussion -> discussion.containsKey(Constants.TAGGED_USER))
+                    .collect(Collectors.toMap(
+                            discussion -> discussion.get(Constants.DISCUSSION_ID).toString(),
+                            discussion -> (List<String>) discussion.get(Constants.TAGGED_USER))));
+        }
```

Comment:
there is a possibility that tagged_user is not provided by the client -- i.e. user is creating the answer for the post itself -- instead of replying to existing answer.